include/picongpu/particles/creation/SpawnFromSourceSpecies.kernel

```diff
@@ -0,0 +1,402 @@
+    {
+        using Modules = T_ModuleConfig;
+        using TypeNumber = T_TypeNumber;
+
+        /** call operator
+         *
+         * called by picongpu stage/sub-stage
+         *
+         * @param worker object containing the device and block information,
+         *  passed by PMACC_KERNEL call
+         * @param areMapping mapping of blockIndex to block superCell index
+         * @param sourceBox deviceDataBox giving access to the particle frames of sourceSpecies particles of all local
+         *  superCells
+         * @param productBox deviceDataBox giving access to the particle frames of productSpecies particles of all
+         * local superCells
+         * @param sharedDataBoxes deviceDataBoxes giving access to the some superCell/Cell/global data required by the
+         *  modul functors or to which the functors write
+         */
+        template<
+            typename T_Worker,
+            typename T_AreaMapping,
+            typename T_SourceParticleBox,
+            typename T_ProductParticleBox,
+            typename... T_SharedDataBoxes>
+        HDINLINE void operator()(
+            T_Worker const& worker,
+            T_AreaMapping const areaMapping,
+            T_SourceParticleBox sourceBox,
+            T_ProductParticleBox productBox,
+            T_SharedDataBoxes... sharedDataBoxes) const
+        {
+            using SourceSpeciesFramePtr = typename T_SourceParticleBox::FramePtr;
+            using ProductSpeciesFramePtr = typename T_ProductParticleBox::FramePtr;
+
+            // perform sanity checks that input is consistent with assumptions
+            using SanityCheck = typename Modules::template SanityCheckInputs<
+                T_SourceParticleBox,
+                T_ProductParticleBox,
+                T_KernelConfigOptions...,
+                T_SharedDataBoxes...>;
+
+            // get frame size
+            PMACC_CASSERT_MSG(
+                sourceSpecies_framesize_must_less_or_equal_productSpecies_framesize,
+                T_SourceParticleBox::frameSize == T_ProductParticleBox::frameSize);
+            constexpr uint32_t frameSize = T_SourceParticleBox::frameSize;
+
+            // for accessing particle frame list of superCell
+            pmacc::DataSpace<picongpu::simDim> const superCellIndex
+                = areaMapping.getSuperCellIndex(worker.blockDomIdxND());
+
+            using IndexFunctor = typename Modules::template SharedDataBoxIndexFunctor<T_KernelConfigOptions...>;
+            auto const sharedDataBoxIndex = IndexFunctor::getIndex(areaMapping, superCellIndex);
+
+            SourceSpeciesFramePtr sourceSpeciesFrame = sourceBox.getLastFrame(superCellIndex);
+
+            // test for skipping superCell due to no sourceSpecies particles or Skip Test
+            bool const skipSuperCell = Modules::template SkipSuperCellFunctor<T_KernelConfigOptions...>::test(
+                superCellIndex,
+                sharedDataBoxIndex,
+                sharedDataBoxes...);
```

Comment:
insead of 
`sharedDataBoxes...`
use perfectly forwarding
```C++
std::forward<T_SharedDataBoxes>(sharedDataBoxes)...
```

Reply:
Done